### DISCUSSION TITLE
sentry_transaction_set_data() - Discarded unknown attribute

### DISCUSSION THREAD

**Discussion by AntonTenstar** [getsentry/sentry-native#981] [Q&A] [answered]
Hello trying to send custom data to with a transaction, but guess I'm doing something wrong. Anyone got any ideas?

Code:
```C
sentry_transaction_context_t* tx_ctx = sentry_transaction_context_new("test transaction", "performance");
sentry_transaction_t* tx = sentry_transaction_start(tx_ctx, sentry_value_new_null());
sentry_transaction_set_data(tx, "performance", sentry_value_new_int32(1337));
sentry_transaction_finish(tx);
```

Result:
![image](https://github.com/getsentry/sentry-native/assets/90307454/39f87f95-e98d-4b45-a8b7-c4947e22f0c3)

**Comment by supervacuus** [comment]
Hi @AntonTenstar! No, you're not doing anything wrong here, afaics. However, this seems to be a blind spot in the event protocol behind transactions. 

The initial definition from the docs considers transactions to be events that are also spans (== the root span of a transaction):

https://develop.sentry.dev/sdk/event-payloads/transaction/#anatomy
https://docs.sentry.io/product/sentry-basics/concepts/tracing/distributed-tracing/#transactions

This is also how it is implemented in the Native SDK. In that sense, transaction events should have a `data` attribute, because spans do. However, the static types used in most tracing data model implementations do not accept this. I am currently discussing with other developers what the right approach should be here.

**Reply by supervacuus** [reply, marked as answer]
Thanks again for raising this, @AntonTenstar! I will provide a fix for this with the next release. You can track  the changes via #983 and #986.

**Reply by supervacuus** [reply]
We released the fix to this with [0.7.3](https://github.com/getsentry/sentry-native/releases/tag/0.7.3).
